ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java

```diff
@@ -171,6 +192,26 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
             Optional.empty()));
   }
 
+  private boolean validateWithdrawals(final EnginePayloadAttributesParameter payloadAttributes) {
+    // TODO Withdrawals move this into mergeCoordinator.updateForkChoice with the other
+    // payloadAttributes validation?
+    final List<Withdrawal> withdrawals =
```

Comment:
Why do we need to convert this to a domain object if we are just checking that there are withdrawals present? Is this for later when withdrawals will be used?

Reply:
The problem here is that I want shanghaiDefinition to wire in a validator class, but it can't be in the api module due to a circular dependency with core.

So WithdrawalsValidator can't accept WithdrawalParameter

Reply:
Thinking about this more I think this is ok. The Withdrawals will be needed to prepare a payload anyway in the FCU. And the MainnetBlockValidator will also need the WithdrawalsValidator.